plug/pip_resolve.py

```diff
@@ -45,13 +45,19 @@ def create_children_recursive(root_package, key_tree):
                 sys.exit(msg)
         children_packages_as_dist = key_tree[root_name]
         for child_dist in children_packages_as_dist:
+            if child_dist.project_name.lower() in ancestors:
+                continue
+
             child_package = {
                 NAME: child_dist.project_name,
                 VERSION: child_dist.installed_version,
                 FROM: root_package[FROM] +
                       [child_dist.key + VERSION_SEPARATOR + child_dist.installed_version]
             }
-            create_children_recursive(child_package, key_tree)
+
+            ancestors = ancestors.copy()
```

Comment:
I understand this works, but would rather not re-assign `ancestors` with a copy of itself? Maybe something like
```
create_children_recursive(child_package, key_tree, set(ancestors.copy().add(root_name)))
```

Not 100% sure this is better, but it is really just a readability issue. Feel free to ignore.